File: 2016/src/day03/task2/Task2.java
package day03.task2;

import java.nio.file.Files;
import java.nio.file.Path;
import java.time.Duration;
import java.time.Instant;

public class Task2 {
   Path path = Path.of("inputs", "day03", "input.txt");

   void task() {
      try {
         var result = 0;
         var lines = Files.readAllLines(this.path);
         for (var i = 0; i < lines.size(); i += 3) {
            var aSides = lines.get(i).trim().split(" +");
            var bSides = lines.get(i + 1).trim().split(" +");
            var cSides = lines.get(i + 2).trim().split(" +");

            for (var j = 0; j < 3; j++) {
               if (Triangle.isValid(aSides[j], bSides[j], cSides[j])) {
                  result += 1;
               }
            }
         }

         System.out.println(result);
      } catch (Exception e) {
         System.err.println(e);
         System.exit(1);
      }
   }

   public static void main(String[] args) {
      var start = Instant.now();

      new Task2().task();

      System.out.println("Done in " + Duration.between(start, Instant.now()).toString().substring(2).toLowerCase());
   }
}

class Triangle {
   static boolean isValid(String a, String b, String c) {
      var sideA = Integer.parseInt(a);
      var sideB = Integer.parseInt(b);
      var sideC = Integer.parseInt(c);
      return (sideA + sideB) > sideC && (sideB + sideC) > sideA && (sideC + sideA) > sideB;
   }
}

File: 2017/src/day01/task2.ts
import path from "node:path";

const PATH = path.join("inputs", "day01", "input.txt");

async function task2() {
   const input = (await Bun.file(PATH).text()).trim();

   let result = 0;
   for (let i = 0; i < input.length; i++) {
      if (input[i] == input[(i + input.length / 2) % input.length]) {
         result += parseInt(input[i]!);
      }
   }

   console.log(result);
}

console.time("Task Done");
await task2();
console.timeEnd("Task Done");
